### DISCUSSION TITLE
Issue on exposing a multipart/form-data Open API specification through the custom-plugin lib

### DISCUSSION THREAD

**Discussion by marco-basso** [mia-platform/community#205] [Q&A] [answered]
I tried to expose a POST with `'content-type': 'multipart/form-data'` using the `custom-plugin-lib`.

I added the following service options in order to use the binary format on validation:
```
const serviceOptions = {
  ajv: {
    plugins: {
      'ajv-formats': {
        formats: ['binary'],
      },
    },
  },
}
```

And exposed the following request body schema:
```
{
  additionalProperties: false,
  properties: {
    file: {
      description: 'Binary content of the file to upload',
      format: 'binary',
      type: 'string',
    },
  },
  required: ['file'],
  type: 'object',
}
```

when I try to call the exposed endpoint with a file I get this validation error:

```
{
  "error": "Bad Request",
  "message": "body.file must be string",
  "statusCode": 400
}
```

Do someone had this kind of issue and found a workaround on dealing with files?

**Comment by edocrippaofficial** [comment, marked as answer]
Hi, a workaround would be skipping entirely the validation of the body (only for that specific endpoint).

This should do the trick
```
function setValidatorCompiler(fastify) {
  const defaultFastifyAjv = new Ajv({ removeAdditional: true, useDefaults: true, coerceTypes: true })
  fastify.setValidatorCompiler(({ schema, httpPart }) => {
    if (httpPart === 'body') {
      return () => true
    }
    return defaultFastifyAjv.compile(schema)
  })
}
```

**Reply by marco-basso** [reply]
I have found another workaround that may do the trick for me. If I avoid to insert the `type: string` on the validation schema then I do not get the issue. BUT the drawback is that the swagger does not let me upload a file from its interface

**Reply by marco-basso** [reply]
@EdoCrippaOfficial thanks for your support, I finally understood your solution and used it to solve my issue!

**Reply by edocrippaofficial** [reply]
Glad to hear that @marco-basso!
Just remember that this is just a workaround, an that coming up with the actual solution to the problem is better for all of us 😊

**Comment by davidebianchi** [comment]
Hi @marco-basso, can you add an issue on the [custom-plugin-lib](https://github.com/mia-platform/custom-plugin-lib/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)?
I think this could be a bug to resolve on the library.

**Reply by marco-basso** [reply]
Done!
